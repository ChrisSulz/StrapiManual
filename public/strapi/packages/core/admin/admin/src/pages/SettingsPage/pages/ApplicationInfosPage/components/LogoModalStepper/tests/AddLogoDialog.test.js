import React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { fireEvent, render as renderTL } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import AddLogoDialog from '../AddLogoDialog';

const render = (props) =>
  renderTL(
    <ThemeProvider theme={lightTheme}>
      <IntlProvider locale="en" messages={{}} textComponent="span">
        <AddLogoDialog goTo={jest.fn} onClose={jest.fn} setLocalImage={jest.fn} {...props} />
      </IntlProvider>
    </ThemeProvider>
  );

describe('ApplicationInfosPage | AddLogoDialog', () => {
  it('shoud render from computer tab and match snapshot', () => {
    const { container, getByText } = render();

    expect(getByText('Drag and Drop here or')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('shoud render from url tab and match snapshot', () => {
    const { container, getByRole } = render();

    fireEvent.click(getByRole('tab', { name: 'From url' }));

    expect(getByRole('textbox', { name: 'URL' })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
