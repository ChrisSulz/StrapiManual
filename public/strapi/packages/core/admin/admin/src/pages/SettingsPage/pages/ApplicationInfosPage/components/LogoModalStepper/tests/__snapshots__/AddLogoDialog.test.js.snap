// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationInfosPage | AddLogoDialog shoud render from computer tab and match snapshot 1`] = `
.c30 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c5 {
  font-weight: 600;
  font-size: 0.6875rem;
  line-height: 1.45;
  text-transform: uppercase;
  color: #4945ff;
}

.c7 {
  font-weight: 600;
  font-size: 0.6875rem;
  line-height: 1.45;
  text-transform: uppercase;
  color: #666687;
}

.c17 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 1.25;
  color: #32324d;
}

.c23 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #ffffff;
}

.c25 {
  font-size: 0.75rem;
  line-height: 1.33;
  color: #666687;
}

.c0 {
  padding-right: 40px;
  padding-left: 40px;
}

.c2 {
  padding: 16px;
}

.c8 {
  background: #eaeaef;
}

.c10 {
  padding-top: 24px;
  padding-right: 40px;
  padding-bottom: 24px;
  padding-left: 40px;
}

.c12 {
  background: #f6f6f9;
  padding-top: 48px;
  padding-bottom: 32px;
  border-radius: 4px;
  border-style: dashed;
  border-width: 1px;
  border-color: #c0c0cf;
  position: relative;
}

.c14 {
  color: #4945ff;
  width: 3.75rem;
  height: 3.75rem;
}

.c16 {
  padding-top: 12px;
  padding-bottom: 20px;
}

.c19 {
  background: #4945ff;
  padding: 8px;
  padding-right: 16px;
  padding-left: 16px;
  border-radius: 4px;
  border-color: #4945ff;
  border: 1px solid #4945ff;
  cursor: pointer;
}

.c24 {
  padding-top: 24px;
}

.c26 {
  background: #f6f6f9;
  padding-top: 16px;
  padding-right: 20px;
  padding-bottom: 16px;
  padding-left: 20px;
}

.c11 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 8px;
}

.c13 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c20 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c28 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c15 path {
  fill: #4945ff;
}

.c21 {
  position: relative;
  outline: none;
}

.c21 > svg {
  height: 12px;
  width: 12px;
}

.c21 > svg > g,
.c21 > svg path {
  fill: #ffffff;
}

.c21[aria-disabled='true'] {
  pointer-events: none;
}

.c21:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c21:focus-visible {
  outline: none;
}

.c21:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c22 {
  height: 2rem;
}

.c22 svg {
  height: 0.75rem;
  width: auto;
}

.c22[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c22[aria-disabled='true'] .c4 {
  color: #666687;
}

.c22[aria-disabled='true'] svg > g,.c22[aria-disabled='true'] svg path {
  fill: #666687;
}

.c22[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c22[aria-disabled='true']:active .c4 {
  color: #666687;
}

.c22[aria-disabled='true']:active svg > g,.c22[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c22:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c22:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c22 svg > g,
.c22 svg path {
  fill: #ffffff;
}

.c29 {
  height: 2rem;
  border: 1px solid #dcdce4;
  background: #ffffff;
}

.c29 svg {
  height: 0.75rem;
  width: auto;
}

.c29[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c29[aria-disabled='true'] .c4 {
  color: #666687;
}

.c29[aria-disabled='true'] svg > g,.c29[aria-disabled='true'] svg path {
  fill: #666687;
}

.c29[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c29[aria-disabled='true']:active .c4 {
  color: #666687;
}

.c29[aria-disabled='true']:active svg > g,.c29[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c29:hover {
  background-color: #f6f6f9;
}

.c29:active {
  background-color: #eaeaef;
}

.c29 .c4 {
  color: #32324d;
}

.c29 svg > g,
.c29 svg path {
  fill: #32324d;
}

.c9 {
  height: 1px;
  border: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin: 0;
}

.c27 {
  border-radius: 0 0 4px 4px;
  border-top: 1px solid #eaeaef;
}

.c3 {
  border-bottom: 2px solid #4945ff;
}

.c6 {
  border-bottom: 2px solid transparent;
}

.c1[aria-disabled='true'] {
  cursor: not-allowed;
}

.c18 {
  opacity: 0;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 1;
}

<div>
  <div>
    <div
      class="c0"
    >
      <div
        aria-label="How do you want to upload your assets?"
        role="tablist"
      >
        <button
          aria-controls=":r0:-0-tabpanel"
          aria-disabled="false"
          aria-selected="true"
          class="c1"
          id=":r0:-0-tab"
          role="tab"
          tabindex="0"
          type="button"
        >
          <div
            class="c2 c3"
          >
            <span
              class="c4 c5"
            >
              From computer
            </span>
          </div>
        </button>
        <button
          aria-disabled="false"
          aria-selected="false"
          class="c1"
          id=":r0:-1-tab"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <div
            class="c2 c6"
          >
            <span
              class="c4 c7"
            >
              From url
            </span>
          </div>
        </button>
      </div>
      <hr
        class="c8 c9"
      />
    </div>
    <div>
      <div
        aria-labelledby=":r0:-0-tab"
        id=":r0:-0-tabpanel"
        role="tabpanel"
        tabindex="0"
      >
        <form>
          <div
            class="c10"
          >
            <div
              class=""
            >
              <label
                for="logo-upload"
              >
                <div
                  class="c11"
                >
                  <div
                    class="c12 c13"
                  >
                    <svg
                      aria-hidden="true"
                      class="c14 c15"
                      fill="none"
                      height="1rem"
                      viewBox="0 0 24 20"
                      width="1rem"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M21.569 2.398H7.829v1.586h13.74c.47 0 .826.5.826 1.094v9.853l-2.791-3.17a2.13 2.13 0 0 0-.74-.55 2.214 2.214 0 0 0-.912-.196 2.215 2.215 0 0 0-.912.191 2.131 2.131 0 0 0-.74.546l-2.93 3.385-2.973-3.36a2.147 2.147 0 0 0-.741-.545 2.228 2.228 0 0 0-1.824.007c-.286.13-.538.319-.739.553l-2.931 3.432V7.653H2.51v9.894c.023.153.06.304.108.452v.127l.041.095c.057.142.126.28.207.412l.099.15c.074.107.157.207.247.302l.124.119c.13.118.275.222.43.309h.024c.36.214.775.327 1.198.325h16.515c.36-.004.716-.085 1.039-.24.323-.153.606-.375.827-.648a2.78 2.78 0 0 0 .504-.888c.066-.217.108-.44.124-.666V5.078a2.497 2.497 0 0 0-.652-1.81 2.706 2.706 0 0 0-1.776-.87Z"
                        fill="#32324D"
                      />
                      <path
                        d="M12.552 9.199c.912 0 1.651-.71 1.651-1.585 0-.876-.74-1.586-1.651-1.586-.912 0-1.652.71-1.652 1.586 0 .875.74 1.585 1.652 1.585ZM3.303 6.408h.826V3.997h2.477V2.41H4.129V0h-.826c-.219 0-.85.002-.826 0v2.411H0v1.586h2.477v2.41h.826Z"
                        fill="#32324D"
                      />
                    </svg>
                    <div
                      class="c16"
                    >
                      <span
                        class="c4 c17"
                      >
                        Drag and Drop here or
                      </span>
                    </div>
                    <input
                      accept="image/jpeg,image/png,image/svg+xml"
                      class="c18"
                      cursor="pointer"
                      id="logo-upload"
                      name="files"
                      tabindex="-1"
                      type="file"
                    />
                    <button
                      aria-disabled="false"
                      class="c19 c20 c21 c22"
                      type="button"
                    >
                      <span
                        class="c4 c23"
                      >
                        Browse files
                      </span>
                    </button>
                    <div
                      class="c24"
                    >
                      <span
                        class="c4 c25"
                      >
                        Max dimension: 750x750, Max size: 100KB
                      </span>
                    </div>
                  </div>
                </div>
              </label>
            </div>
          </div>
        </form>
        <div
          class="c26 c27"
        >
          <div
            class="c28"
          >
            <div
              class="c20"
            >
              <button
                aria-disabled="false"
                class="c19 c20 c21 c29"
                type="button"
              >
                <span
                  class="c4 c23"
                >
                  Cancel
                </span>
              </button>
            </div>
            <div
              class="c20"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c30"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`ApplicationInfosPage | AddLogoDialog shoud render from url tab and match snapshot 1`] = `
.c25 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c7 {
  font-weight: 600;
  font-size: 0.6875rem;
  line-height: 1.45;
  text-transform: uppercase;
  color: #4945ff;
}

.c5 {
  font-weight: 600;
  font-size: 0.6875rem;
  line-height: 1.45;
  text-transform: uppercase;
  color: #666687;
}

.c23 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #ffffff;
}

.c12 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #32324d;
}

.c0 {
  padding-right: 40px;
  padding-left: 40px;
}

.c2 {
  padding: 16px;
}

.c8 {
  background: #eaeaef;
}

.c10 {
  padding-top: 24px;
  padding-right: 40px;
  padding-bottom: 24px;
  padding-left: 40px;
}

.c20 {
  background: #4945ff;
  padding: 8px;
  padding-right: 16px;
  padding-left: 16px;
  border-radius: 4px;
  border-color: #4945ff;
  border: 1px solid #4945ff;
  cursor: pointer;
}

.c17 {
  background: #f6f6f9;
  padding-top: 16px;
  padding-right: 20px;
  padding-bottom: 16px;
  padding-left: 20px;
}

.c19 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c11 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
}

.c21 {
  position: relative;
  outline: none;
}

.c21 > svg {
  height: 12px;
  width: 12px;
}

.c21 > svg > g,
.c21 > svg path {
  fill: #ffffff;
}

.c21[aria-disabled='true'] {
  pointer-events: none;
}

.c21:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c21:focus-visible {
  outline: none;
}

.c21:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c24 {
  height: 2rem;
}

.c24 svg {
  height: 0.75rem;
  width: auto;
}

.c24[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c24[aria-disabled='true'] .c4 {
  color: #666687;
}

.c24[aria-disabled='true'] svg > g,.c24[aria-disabled='true'] svg path {
  fill: #666687;
}

.c24[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c24[aria-disabled='true']:active .c4 {
  color: #666687;
}

.c24[aria-disabled='true']:active svg > g,.c24[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c24:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c24:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c24 svg > g,
.c24 svg path {
  fill: #ffffff;
}

.c22 {
  height: 2rem;
  border: 1px solid #dcdce4;
  background: #ffffff;
}

.c22 svg {
  height: 0.75rem;
  width: auto;
}

.c22[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c22[aria-disabled='true'] .c4 {
  color: #666687;
}

.c22[aria-disabled='true'] svg > g,.c22[aria-disabled='true'] svg path {
  fill: #666687;
}

.c22[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c22[aria-disabled='true']:active .c4 {
  color: #666687;
}

.c22[aria-disabled='true']:active svg > g,.c22[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c22:hover {
  background-color: #f6f6f9;
}

.c22:active {
  background-color: #eaeaef;
}

.c22 .c4 {
  color: #32324d;
}

.c22 svg > g,
.c22 svg path {
  fill: #32324d;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 {
  height: 1px;
  border: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin: 0;
}

.c16 {
  border: none;
  border-radius: 4px;
  padding-bottom: 0.65625rem;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 0.65625rem;
  color: #32324d;
  font-weight: 400;
  font-size: 0.875rem;
  display: block;
  width: 100%;
  background: inherit;
}

.c16::-webkit-input-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c16::-moz-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c16:-ms-input-placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c16::placeholder {
  color: #8e8ea9;
  opacity: 1;
}

.c16[aria-disabled='true'] {
  color: inherit;
}

.c16:focus {
  outline: none;
  box-shadow: none;
}

.c15 {
  border: 1px solid #dcdce4;
  border-radius: 4px;
  background: #ffffff;
  outline: none;
  box-shadow: 0;
  -webkit-transition-property: border-color,box-shadow,fill;
  transition-property: border-color,box-shadow,fill;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
}

.c15:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c18 {
  border-radius: 0 0 4px 4px;
  border-top: 1px solid #eaeaef;
}

.c6 {
  border-bottom: 2px solid #4945ff;
}

.c3 {
  border-bottom: 2px solid transparent;
}

.c1[aria-disabled='true'] {
  cursor: not-allowed;
}

<div>
  <div>
    <div
      class="c0"
    >
      <div
        aria-label="How do you want to upload your assets?"
        role="tablist"
      >
        <button
          aria-disabled="false"
          aria-selected="false"
          class="c1"
          id=":r2:-0-tab"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <div
            class="c2 c3"
          >
            <span
              class="c4 c5"
            >
              From computer
            </span>
          </div>
        </button>
        <button
          aria-controls=":r2:-1-tabpanel"
          aria-disabled="false"
          aria-selected="true"
          class="c1"
          id=":r2:-1-tab"
          role="tab"
          tabindex="0"
          type="button"
        >
          <div
            class="c2 c6"
          >
            <span
              class="c4 c7"
            >
              From url
            </span>
          </div>
        </button>
      </div>
      <hr
        class="c8 c9"
      />
    </div>
    <div>
      <div
        aria-labelledby=":r2:-1-tab"
        id=":r2:-1-tabpanel"
        role="tabpanel"
        tabindex="0"
      >
        <form>
          <div
            class="c10"
          >
            <div>
              <div
                class=""
              >
                <div
                  class="c11"
                >
                  <label
                    class="c4 c12 c13"
                    for=":r4:"
                  >
                    URL
                  </label>
                  <div
                    class="c14 c15"
                  >
                    <input
                      aria-disabled="false"
                      aria-invalid="false"
                      aria-required="false"
                      class="c16"
                      id=":r4:"
                      name="logo-url"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c17 c18"
          >
            <div
              class="c14"
            >
              <div
                class="c19"
              >
                <button
                  aria-disabled="false"
                  class="c20 c19 c21 c22"
                  type="button"
                >
                  <span
                    class="c4 c23"
                  >
                    Cancel
                  </span>
                </button>
              </div>
              <div
                class="c19"
              >
                <button
                  aria-disabled="false"
                  class="c20 c19 c21 c24"
                  type="submit"
                >
                  <span
                    class="c4 c23"
                  >
                    Next
                  </span>
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    class="c25"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
